packages/ui/src/SelectField.tsx

```diff
@@ -31,6 +31,50 @@ const Input = (props: any) => {
   return <components.Input {...props} autoComplete="chrome-off" />
 }
 
+// This component is implemented because we want to style things with the css (as opposed to the react-select
+// recommended way of styling things via js).
+//
+// We simply inject `.multiValueIsFocused`.
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+const MultiValue = (props: MultiValueProps<any>) => {
```

Comment:
nit: SHould `MultiValue` come after `MultiValueRemove`and `MultiValueLabel` since it references them? AFAIU, `consts` are not hoisted all the way to the top, so it works only because these arrow functions are converted to regular functions during transpilation

Reply:
i don't get the point here??? MultiValue uses the other 2 components inside its body - so they will be referenced at the time of the call. declaration order should not be an issue here in any case.

Reply:
@utkukaratas here is a simple reproducer https://www.typescriptlang.org/play?noUnusedLocals=true&noUnusedParameters=true&declaration=false&target=99#code/MYewdgzgLgBAhjAvDAFASiQPhgbwFAwwBG6eAvnnKXqJLEUqhotrRCADYCmAdByAHMUAcgAWAW0nC0QA
Try running it and see the console

Reply:
We get away with it because webpack transpiles it to https://www.typescriptlang.org/play?noUnusedLocals=true&noUnusedParameters=true&declaration=false&target=99#code/PTAEGMHsDsGcBdQENQF5QAoCUaB8oBvAKBFDICNsSwBfa5K+qORctTHVfZ2SAGwCmAOj6QA5hgDkACwC28yViL0AKgCckcAA4BLQQBNQ5AJ6gA7gPJak4ANah4kZQDMArtHDwdMZB0JEKKjokRjcPLx9KHGIyHn5hUQkZBSUaIA

Reply:
ok calling a() before declaring "b" might of course be problematic but in our case noone is calling MultiValue before declaring the other two. and MultiValue is instantiated much much later, in fact ages later, when SelectField is instantiated somewhere in the app code.

Reply:
@utkukaratas yeah, you are right. Here is a great article that helped me to brush up my knowledge about lexical scoping BTW https://blog.bitsrc.io/hoisting-in-modern-javascript-let-const-and-var-b290405adfda

Reply:
@aigoncharov you're a review machine dude, and every machine needs a reboot at one point, you just got segfaulted :D